inc/html.class.php

```diff
@@ -4513,11 +4513,35 @@ static function jsAdaptDropdown($id, $params = []) {
       $templateresult    = $params["templateResult"] ?? "templateResult";
       $templateselection = $params["templateSelection"] ?? "templateSelection";
 
+      if (!isset($params['dropdownParentSelector'])) {
+         $params['dropdownParentSelector'] = 'auto';
+      }
+      switch ($params['dropdownParentSelector']) {
+         case 'auto':
+            $dropdown_parent_value = '$("#'.$id.'").closest("div.modal, body")';
+            break;
+         case 'document.body':
+            $dropdown_parent_value = '$(document.body)';
+            break;
+         default:
+            $dropdown_parent_value = '$('.json_encode($params['dropdownParentSelector']).')';
+      }
+
+      $search_autofocus_event = <<<JS
+      $(select2_el).on('select2:open', (e) => {
+         const select2_selector = `.select2-search__field[aria-controls='select2-\${e.target.id}-results']`;
+         $(select2_selector).each((k, v) => {
+            v.focus();
+         });
```

Comment:
I guess selector will return a unique result, so it can be simplified to:
```suggestion
         $(`.select2-search__field[aria-controls='select2-\${e.target.id}-results']`).focus();
```